pkg/network/protocols/http/event_pipeline_test.go

```diff
@@ -0,0 +1,143 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+//go:build linux_bpf
+
+package http
+
+import (
+	"bytes"
+	"encoding/binary"
+	"runtime"
+	"testing"
+	"time"
+	"unsafe"
+
+	"github.com/DataDog/datadog-agent/pkg/network/config"
+	"github.com/DataDog/datadog-agent/pkg/network/protocols/events"
+	"github.com/stretchr/testify/require"
+)
+
+// structToBytes serializes the provided events into a byte array.
+func structToBytes(b *testing.B, events []EbpfEvent, numOfEventsInBatch int) [4096]int8 {
+	var result [4096]int8
+	var buffer bytes.Buffer
+
+	// Serialize the events in the slice
+	for i := 0; i < numOfEventsInBatch; i++ {
+		// Use modulo to cycle through the provided events
+		event := events[i%len(events)]
+		require.NoError(b, binary.Write(&buffer, binary.LittleEndian, event))
+	}
+
+	serializedData := buffer.Bytes()
+	// Ensure the serialized data fits into the result array
+	require.LessOrEqualf(b, len(serializedData), len(result), "serialized data exceeds 4096 bytes")
+
+	for i, b := range serializedData {
+		result[i] = int8(b)
+	}
+
+	return result
+}
+
+// setupBenchmark sets up the benchmark environment by creating a consumer, protocol, and configuration.
+func setupBenchmark(b *testing.B, c *config.Config, totalEventsCount int) (*events.Consumer[EbpfEvent], *protocol) {
+	// Serialized data can't exceed 4096 bytes that why we can insert 14 events in a batch.
+	const numOfEventsInBatch = 14
+
+	program, err := events.NewEBPFProgram(c)
+	require.NoError(b, err)
+
+	httpTelemetry := NewTelemetry("http")
+
+	p := protocol{
+		cfg:        c,
+		telemetry:  httpTelemetry,
+		statkeeper: NewStatkeeper(c, httpTelemetry, NewIncompleteBuffer(c, httpTelemetry)),
+	}
+	consumer, err := events.NewConsumer("test", program, p.processHTTP)
+	require.NoError(b, err)
+
+	go generateMockEvents(b, c, consumer, numOfEventsInBatch, totalEventsCount)
```

Comment:
Could you also ensure that this routine completes before asserting in the test? This might help prevent race conditions